test/sqlancer/TestComparatorHelper.java

```diff
@@ -0,0 +1,54 @@
+package sqlancer;
+
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+public class TestComparatorHelper {
+    // TODO: Implement tests for the other ComparatorHelper methods
+
+    @Test
+    public void testAssumeResultSetsAreEqualWithEqualSets() {
+        List<String> r1 = Arrays.asList("a", "b", "c");
+        List<String> r2 = Arrays.asList("a", "b", "c");
+        ComparatorHelper.assumeResultSetsAreEqual(r1, r2, "", Arrays.asList(""), null);
+
+    }
+
+    @Test
+    public void testAssumeResultSetsAreEqualWithUnequalLengthSets() {
+        List<String> r1 = Arrays.asList("a", "b", "c");
+        List<String> r2 = Arrays.asList("a", "b", "c", "d", "g");
+        // NullPointerException is raised instead of AssertionError because state is null and the state.getState()...
+        // line occurs before AssertionError is thrown, but it's good enough as an indicator that one of the Exceptions
+        // is raised
+        assertThrowsExactly(NullPointerException.class, () -> {
```

Comment:
I feel like checking for any kind of `Throwable` would be more general here, but specifically checking for a `NullPointerException` is also fine.

Reply:
I see! It might still be debatable whether this is a part of the method's contract (i.e., its API) or an implementation artifact. Either case is fine - at most, we have to update the unit test if the implementation ever changes.

Reply:
ah, makes sense!